kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/marshalling/MarshallingFormatTest.java

```diff
@@ -0,0 +1,57 @@
+package org.kie.server.api.marshalling;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+
+public class MarshallingFormatTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void testEmptyMarshallingFormat() {
+        exception.expectMessage("Invalid marshalling format []");
+        MarshallingFormat.fromType("");
+    }
+
+    @Test
+    public void testNullMarshallingFormat() {
+        exception.expectMessage("Invalid marshalling format [null]");
+        MarshallingFormat.fromType(null);
+    }
+
+    @Test
+    public void testExpectedMarshallingFormats() {
+        assertEquals(MarshallingFormat.JSON, MarshallingFormat.fromType("json"));
+        assertEquals(MarshallingFormat.JAXB, MarshallingFormat.fromType("xml"));
+        assertEquals(MarshallingFormat.XSTREAM, MarshallingFormat.fromType("xstream"));
+
+        assertEquals(MarshallingFormat.JSON, MarshallingFormat.fromType("application/json"));
+        assertEquals(MarshallingFormat.JAXB, MarshallingFormat.fromType("application/xml"));
+        assertEquals(MarshallingFormat.XSTREAM, MarshallingFormat.fromType("application/xstream"));
+    }
+
+    @Test
+    public void testMarshallingFormatsWithExtraneousParameters() {
+        assertEquals(MarshallingFormat.JSON, MarshallingFormat.fromType("application/json;"));
+        assertEquals(MarshallingFormat.JAXB, MarshallingFormat.fromType("application/xml;"));
+        assertEquals(MarshallingFormat.XSTREAM, MarshallingFormat.fromType("application/xstream;"));
+        assertEquals(MarshallingFormat.JSON, MarshallingFormat.fromType("application/json;encode="));
+        assertEquals(MarshallingFormat.JAXB, MarshallingFormat.fromType("application/xml;encode=utf-8"));
+        assertEquals(MarshallingFormat.XSTREAM, MarshallingFormat.fromType("application/xstream;utf-8"));
+    }
+
+    @Test
+    public void testMarshallingFormatCase() {
+        assertEquals(MarshallingFormat.JSON, MarshallingFormat.fromType("JSON"));
+    }
+
+    @Test
+    public void testEdgeCaseWithJaxb() {
+        exception.expectMessage("Invalid marshalling format [jaxb]");
+        MarshallingFormat.fromType("jaxb");
+    }
```

Comment:
correct, that would have worked. Though the special content type header should only be used for xstream as other types should rely on default application/json or application/xml